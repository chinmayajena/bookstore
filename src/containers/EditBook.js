import React, { Component } from "react";
import CreateBookForm from "components/CreateBookForm.js";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { updateBook } from "actions/actionCreators.js";
import { bindActionCreators } from "redux";
import ErrorBoundary from "components/ErrorBoundary.js";

class EditBook extends Component {
  handleFormSubmit = values => {
    const { updateBook } = this.props;
    updateBook(values);
    this.props.history.push("/");
  };

  render() {
    return (
      <ErrorBoundary>
        <CreateBookForm
          initialValues={this.props.initialValues}
          onSubmit={this.handleFormSubmit}
        />
      </ErrorBoundary>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateBook
    },
    dispatch
  );
};

const getRecordForUpdate = (booksArray, recordId = 1) => {
  let records = booksArray.filter(book => book.id === recordId);
  return records.length > 0 ? records[0] : {};
};

export default connect(
  state => ({
    initialValues: getRecordForUpdate(
      state.books,
      state.visibilityFilter.updateRecordId
    )
  }),
  mapDispatchToProps
)(EditBook);
