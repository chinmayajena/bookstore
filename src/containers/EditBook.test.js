import React from "react";
import { shallow } from "enzyme";
import { EditBook } from "./EditBook.js";
import configureStore from "redux-mock-store";

describe("<EditBook />", () => {
  const initialState = {};
  const mockStore = configureStore();
  const mockSubmitfn = jest.fn();
  let store, component;

  beforeEach(() => {
    store = mockStore(initialState);
    component = shallow(<EditBook updateBook={mockSubmitfn} store={store} />);
  });

  it("renders the component", () => {
    expect(component.debug()).toMatchSnapshot();
    const wrapper = component.find("ErrorBoundary");
    expect(wrapper.length).toBe(1);
  });
});
