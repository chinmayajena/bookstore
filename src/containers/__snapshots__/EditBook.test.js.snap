// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditBook /> renders the component 1`] = `
"<ErrorBoundary>
  <ReduxForm initialValues={[undefined]} onSubmit={[Function]} touchOnBlur={true} touchOnChange={false} persistentSubmitErrors={false} destroyOnUnmount={true} shouldAsyncValidate={[Function: defaultShouldAsyncValidate]} shouldValidate={[Function: defaultShouldValidate]} shouldError={[Function: defaultShouldError]} shouldWarn={[Function: defaultShouldWarn]} enableReinitialize={true} keepDirtyOnReinitialize={true} updateUnregisteredFields={true} getFormState={[Function: getFormState]} pure={true} forceUnregisterOnUnmount={false} form=\\"createBook\\" />
</ErrorBoundary>"
`;
